web/app/view/dialog/AttributeController.js

```diff
@@ -39,5 +39,49 @@ Ext.define('Traccar.view.dialog.AttributeController', {
             record.save();
         }
         button.up('window').close();
+    },
+
+    onValidityChange: function (form, valid) {
+        this.lookupReference('saveButton').setDisabled(!valid);
+    },
+
+    defaultFieldConfig: {
+        name: 'value',
+        reference: 'valueField',
+        allowBlank: false,
+        fieldLabel: Strings.stateValue
+    },
+
+    onNameChange: function (combobox, newValue) {
+        var type, config, valueField = this.lookupReference('valueField'),
+                attribute = combobox.getStore().getById(newValue);
```

Comment:
I think it's better to split it into separate line. It would make it more readable.